src/app/content/highlights/components/cardUtils.ts

```diff
@@ -0,0 +1,71 @@
+import { Highlight } from '@openstax/highlighter';
+import { HTMLElement, } from '@openstax/types/lib.dom';
+import React from 'react';
+import { findElementSelfOrParent } from '../../../domUtils';
+import { assertWindow } from '../../../utils';
+
+export const useDebouncedWindowSize = () => {
+  const window = assertWindow();
+  const timeout = React.useRef(0);
+  const [size, setSize] = React.useState([0, 0]);
+
+  React.useLayoutEffect(() => {
+    const updateSize = () => {
+      clearTimeout(timeout.current);
+      timeout.current = setTimeout(() => {
+        setSize([window.innerWidth, window.innerHeight]);
+      }, 50);
+    };
+    window.addEventListener('resize', updateSize);
+    updateSize();
```

Comment:
Maybe initialize the state with current width and height instead of calling the `updateSize` manually?